src/Http/Senders/GuzzleSender.php

```diff
@@ -172,6 +172,10 @@ protected function createRequestOptions(PendingRequest $pendingRequest): array
             $requestOptions[$configVariable] = $value;
         }
 
+        if ($pendingRequest->delay()->all() > 0) {
```

Comment:
Could you use `isNotEmpty()` here instead? This follows the convention of the body repositories?

Reply:
I did this on purpose to avoid sleep exceptions from negative integers. Now we check if the delay is >0 and if not there is no need for delaying the request.